akvo/rsr/views/py_reports/nuffic_country_level_map_report.py

```diff
@@ -19,6 +19,68 @@
 
 from . import utils
 
+REPORT_NAME = 'nuffic_country_level_map_report'
+
+
+@login_required
+def add_email_report_job(request, program_id):
+    program = get_object_or_404(Project, pk=program_id)
+    country = request.GET.get('country', '').strip()
+    if not country:
+        return HttpResponseBadRequest('Please provide the country code!')
+    show_comment = request.GET.get('comment', '').strip()
+    start_date = request.GET.get('period_start', '').strip()
+    end_date = request.GET.get('period_end', '').strip()
+    return utils.add_email_report_job(
+        report=REPORT_NAME,
+        payload={
+            'program_id': program.id,
+            'country': country,
+            'show_comment': show_comment,
+            'start_date': start_date,
+            'end_date': end_date,
+        },
+        recipient=request.user.email
+    )
+
+
+def handle_email_report(params, recipient):
+    country = params.get('country')
+    show_comment = True if params.get('comment', '') == 'true' else False
+    start_date = utils.parse_date(params.get('period_start', ''), datetime(1900, 1, 1))
+    end_date = utils.parse_date(params.get('period_end', ''), datetime(2999, 12, 31))
+
+    country = Country.objects.get(iso_code=country)
+    program = Project.objects.get(pk=params['program_id'])
+
+    project_ids = program.descendants()\
+        .exclude(pk=program.id)\
+        .exclude(Q(title__icontains='test') | Q(subtitle__icontains='test'))\
```

Comment:
:thinking: does nuffic only have one program and it will never contain test?

Reply:
At the time this feature was requested, nuffic had several test projects that they didn't want to show in the report. So far, there has been no request from them to negate it.

The query should work even if nuffic has more than one program. It should also work for other partner programs if enabled.